### PR TITLE
feat: support property editing

### PR DESCRIPTION
**Pull request by veselov**
Resolves !6

Please LMK if you see any issues with the change.
I've refactored the add dependency related classes to have the basic "I modify POM files" piece move out to corresponding base classes.

### CODE REVIEW

**Comment by dearrudam** [comment]
@veselov thank you so much for this amazing contribution!!! 👏👏👏

File: src/main/java/br/org/soujava/pomeditor/api/Property.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ * limitations under the License.
+ *
+ */
+
+package br.org.soujava.pomeditor.api;
+
+import java.util.Objects;
+
+/**
+ * Represents a dependency to be added
```
Comment (dearrudam) [inline src/main/java/br/org/soujava/pomeditor/api/Property.java:23]:
```suggestion
 * Represents a property to be added, or changed or removed
```

File: src/main/java/br/org/soujava/pomeditor/mojo/EditingMojo.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package br.org.soujava.pomeditor.mojo;
+
+import br.org.soujava.pomeditor.api.PomChange;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
```
Comment (dearrudam) [inline src/main/java/br/org/soujava/pomeditor/mojo/EditingMojo.java:27]: @veselov could you add a javadoc documentation for this abstract class explaining its purposes, please?

**Review by dearrudam — CHANGES_REQUESTED** [review]

**Comment by veselov** [comment]
@dearrudam For the new files, do you expect copyright year to be set to current?

**Comment by veselov** [comment]
@dearrudam I also didn't see any rules on squashing during PR reviews, so I did squash as I didn't see any commit keywords to use for code reviews, but LMK if you'd like me to un-squash it.

**Comment by dearrudam** [comment]
> @dearrudam I also didn't see any rules on squashing during PR reviews, so I did squash as I didn't see any commit keywords to use for code reviews, but LMK if you'd like me to un-squash it.

By now, there are no worries about squashing commits, so no worries about that for now! Everything is LGTM! :+1:

**Comment by dearrudam** [comment]
> @dearrudam For the new files, do you expect copyright year to be set to current?

No, but if you want to add the current year on each new file that you have added, feel free!

File: src/main/java/br/org/soujava/pomeditor/mojo/EditingMojo.java (hunk 2 above)
Comment (dearrudam) [reply src/main/java/br/org/soujava/pomeditor/mojo/EditingMojo.java:27]: Great!!! Thanks!

File: src/main/java/br/org/soujava/pomeditor/api/Property.java (hunk 1 above)
Comment (veselov) [reply src/main/java/br/org/soujava/pomeditor/api/Property.java]: That's done as well.

**Review by dearrudam — APPROVED** [review]
@veselov, congratulations :clap::clap::clap:, and thank you so much for the contribution :+1: !!! I'll merge it into the project!!! Welcome to the contributor's board :rocket:!!!

**Comment by dearrudam** [comment]
I'll be releasing a new version soon...I'll keep you updated!

**Comment by veselov** [comment]
@dearrudam You are very welcome! Thank you for maintaining this project in the first place.

**Pull request merged**